src/plugins/mer/meremulatordevice.cpp

```diff
@@ -722,23 +784,41 @@ void MerEmulatorDeviceManager::onDeviceRemoved(Core::Id id)
 {
     m_deviceSshPortCache.remove(id);
     m_deviceQmlLivePortsCache.remove(id);
+    m_deviceMemorySizeCache.remove(id);
+    m_deviceCpuCountCache.remove(id);
+    m_vdiCapacityCache.remove(id);
 
     emit storedDevicesChanged();
 }
 
 void MerEmulatorDeviceManager::onDeviceListReplaced()
 {
+    QProgressDialog progress(tr("Applying virtual machine settings"), tr("Abort"), 0, 0, ICore::dialogParent());
+    progress.setWindowModality(Qt::WindowModal);
+    QPushButton cancelButton(tr("Abort"));
+    cancelButton.setDisabled(true);
+    progress.setCancelButton(&cancelButton);
+    progress.show();
```

Comment:
I don't set QProgressDialog::minimumDuration because we can't correctly cancel resize vdi operation

Reply:
Unless I missed something there is no conflict between setting minimumDuration and inability to cancel. The minimumDuration serves just as a mean to delay displaying the progress dialog. It is set by default - to 4000ms. I would prefer setting 2000ms instead, because we want to show the message that the operation was successfully done at the end, and 4s feels quite a long period of seamless inactivity.

Reply:
Hmm. In this function no asynchronous function and QProgressDialog::minimumDuration don't open dialog automatically, only called explicitly work...

Reply:
Yeah, that's the problem with the blocking call - it newer returns to the event loop and freezes the thread. So if you are sure it will never take longer than a second or two to resize the VDI, just omit the progress dialog here and in MerOptionsWidget. Otherwise it would have to be reimplemented in a non-blocking way.

Reply:
Add asynchronous to MerVirtualBoxManager::setVdiCapacityMb